msgpack-core/src/main/java/org/msgpack/core/buffer/MessageBuffer.java

```diff
@@ -170,22 +177,85 @@
      */
     protected final int size;
 
-    public static MessageBuffer allocate(int length)
+    /**
+     * Reference is used to hold a reference to an object that holds the underlying memory so that it cannot be
+     * released by the garbage collector.
+     */
+    protected final ByteBuffer reference;
+
+    /**
+     * Allocates a new MessageBuffer backed by a byte array.
+     *
+     * @throws IllegalArgumentException If the capacity is a negative integer
+     *
+     */
+    public static MessageBuffer allocate(int size)
     {
-        return wrap(new byte[length]);
+        if (size < 0) {
+            throw new IllegalArgumentException("size must not be negative");
+        }
+        return wrap(new byte[size]);
     }
 
+    /**
+     * Wraps a ByteBuffer into a MessageBuffer.
+     *
+     * The new MessageBuffer will be backed by the given byte array. Modifications to the new MessageBuffer will cause the byte array to be modified and vice versa.
+     *
+     * The new buffer's size will be array.length. hasArray() will return true.
+     *
+     * @param array the byte array that will gack this MessageBuffer
+     * @return
+     *
+     */
     public static MessageBuffer wrap(byte[] array)
     {
         return newMessageBuffer(array, 0, array.length);
     }
 
+    /**
+     * Wraps a ByteBuffer into a MessageBuffer.
+     *
+     * The new MessageBuffer will be backed by the given byte array. Modifications to the new MessageBuffer will cause the byte array to be modified and vice versa.
```

Comment:
well, I copied mof of the text from JavaDoc of ByteBuffer: https://docs.oracle.com/javase/7/docs/api/java/nio/ByteBuffer.html#wrap(byte[])